<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<html>
<head>
    <title>MSA</title>
    <link rel='stylesheet' href='tapu.css' type='text/css' />
</head>
<body>


<p align="center">
<font size="7">Multivariate Data Analysis</font> <br />
<font size="4">Previously known as multivariate statistical analysis</font>
</p>

<hr />

<p>There are only three essential steps here, and the second one is even optional:</p>
<ol>
    <li>Alignment in two dimensions </li>
    <li>Dimension-reduction -- expression of a <i>m</i>x<i>n</i> image 
        using only a few terms, i.e., eigenvectors </li> 
    <li>Classification </li> 
</ol>

<p>
   For the classification below to be sensible, the images will need to 
   have been aligned. The alignment step here is optional if the images 
   have been aligned already. 
</p>

<p>
   The dimension-reduction step is even optional, in theory. In principle, 
   one could classify the raw images (which is what SPIDER operation 
   <a href="../../man/apc.html">'AP C'</a> 
   does). As an example here, I'm using correspondence analysis for 
   the dimension-reduction. A similar method is principal-component 
   analysis (PCA); to run PCA, one needs to change an option under SPIDER 
   operation <a href="../../man/cas.html">'CA S'</a> 
   (here, in the procedure file <a href="ca-pca.msa">ca-pca.msa</a>). 
</p>

<p>
 For classification, there are three methods illustrated here: Diday's 
 method, Ward's method, and K-means. The individual classification 
 operations are described in more depth in the 
 <a href="../classification/tutorial.html">classification tutorial</a>. 
</p>

<hr />

<p><b><font size="4">Getting started</font></b></p>

<ul>
   <li>Download and unpackage the 
   <a href="tar_archive/latest_msa_proc.tar">tarball containing the procedure files</a>      <br />
   The files will be written into the current directory, so create a new one, if so desired. </li>
</ul>

<hr />

<p><b><font size="4">Quick version</font></b></p>

<p>
This quick version is here partly for simplicity, and partly for legacy reasons.  
The more complete and more up-to-date workflow is presented below.
</p>

The overall inputs are simply:
<ul>
    <li>a particle stack</li>
    <li>a selection file, listing the image numbers in the stack file above</li>
</ul>

If your inputs are not in this form, refer to the full alignment below and then return here.

<ol>
    <li><b>[Optional] High-pass filtration</b> 
    <p>For negatively-stained images, it is often the case that the puddle of stain surrounding the particles 
    dominates the classification (although maybe not the case for deep-stained specimens).  
    In this case, it may be helpful to high-pass filter the images using a filter radius 
    larger than the size of your particle.  If you cut off <u>too</u> much of the low-frequency information 
    however, the aligning power of the images will be lost.  You may need to tweak the radii.<br />
    <ul>
        <li>PROCEDURE FILE: <a href="highpass.msa">highpass.msa</a></li> 
        <li>INPUT PARAMETERS: filter radii                         </li>
        <li>INPUTS:  selection file, unfiltered particle stack     </li>
        <li>OUTPUTS: filtered particles                            </li>
    </ul>
    </p> 
    
    <li><b>Reference-free alignment</b>. --  choose one of these two options:
    <ol type="a">
        <li>[Recommended] Pairwise alignment
        <ul>
            <li>PROCEDURE FILE: <a href="pairwise.msa">pairwise.msa</a>       </li> 
            <li>INPUT PARAMETER: object diameter (in pixels, after decimation)</li>
            <li>INPUTS: unaligned particles, selection file                   </li>
            <li>OUTPUTS: aligned particles, averages 
 
            <p>Conceptually, this alignment first aligns pairs of images and averages them.
                 Then, it aligns pairs of averages of those pairs and averages them, and so forth.  
                 This type of alignment appears to be less random than does 
                 <a href="../../man/apsr.html">'AP SR'</a>, 
                 which chooses seed images as alignment references.</p>
            <p><b>Reference</B>: Marco S, Chagoyen M, de la Fraga LG, Carazo JM, Carrascosa JL (1996)
                Ultramicroscopy <b>66</B>: 5-10.</p></li>
        </ul>
        
        <li>Using <a href="../../man/apsr.html">'AP SR'</a> 
        <ul>
            <li>PROCEDURE FILE: <a href="apsr4class.msa">apsr4class.msa</a></li>
            <li>INPUT PARAMETER: object diameter (pixels, after decimation)</li>
            <li>INPUTS: unaligned particle stack, selection file           </li>
            <li>OUTPUTS: aligned particles, averages
 
            <p>There may to be a memory limit in 
            <a href="../../man/apsr.html">'AP SR'</a>.
            If you get a core dump, truncate the selection file and try again.</p></li>
        </ul>

    </ol>

    <li><b>Low-pass filtration and downsampling</b> 
    <p>The low-pass filtration step will make the particles easier to visualize. 
       The downsampling will speed up the computation.
    </p>
    
    <ul>
        <li>PROCEDURE FILE: <a href="filtershrink.msa">filtershrink.msa</a></li> 
        <li>INPUT PARAMETERS: filter radii, decimation factor              </li>
        <li>INPUTS: selection file, unfiltered particle stack              </li>
        <li>OUTPUTS: filtered particles                                    </li>
    </ul>
    <p />

    <li><b>Dimension-reduction</b> 
    <ul>
    <li>1. <i>(optional)</I> Create a custom mask for your particle.
        The alternative is that a circular mask will be used below.</li>
        <ul>
            <li>PROCEDURE FILE: <a href="custommask.msa">custommask.msa</a></li>
            <li>PARAMETERS:                                                </li>
                <ul>
                    <li>Fourier filter radius for input image              </li>
                    <li>first threshold, in units of standard deviations   <br />
                        That is, the threshold will be set to the average intensity 
                        plus some number times the standard deviation.     </li>
                    <li>Fourier filter radius for the thresholded image    <br />
                        The first thresholded image is often too jagged, 
                        so this filtration serves to smooth it.            </li>
                    <li>second threshold, for the filtered mask, between 0 and 1 <br />
                        The lower the number, the more generous the mask will be.</li>
                </ul>
            <li>INPUT: average image, such as <b>pairwise/rfreeavg001</b>  </li>
            <li>OUTPUT: <b>stkmask</B>, whose slices are as follows:       </li>
                <ol>
                    <li>filtered image                </li>
                    <li>thresholded image             </li>
                    <li>filtered mask                 </li>
                    <li>final mask                    </li>
                    <li>mask-multipled image          </li>
                    <li>inverted mask                 </li>
                    <li>inverted mask-multiplied image</li>
                </ol>
            <br />You may need to fiddle with the parameters to get a nice mask.
        </ul>

            <center><p><table border="1">
                <tr><td><center><p align="center">
                <img src="images/custom.gif">         <br />
                slices of the custom-made mask stack file
                </p></center></td></tr>
            </table></p></center>

    <li>2. Run correspondence analysis or principal component analysis</li>
        <ul>
            <li>PROCEDURE FILE: <a href="ca-pca.msa">ca-pca.msa</a>   </li>
            <li>uses Python script <a href="eigendoc.py">eigendoc.py</a> (J.S.L. &amp; T.R.S.) </li>
            <li>INPUT PARAMETER: number of eigenfactors to calculate (more than 99 will require 
                                 some user modification)              </li>
            <li>INPUT: aligned particles                              </li>
            <li>OUTPUTS:                                              </li>
              <UL type="circle">
                <li><a href="images/factormap.gif">factormaps</a> 
                  -- 2D plot of a pair of factors (e.g., 1 vs. 2) for each image </li>
                <li><a href="images/histogram.gif">eigenvalue histogram</a> 
                  -- shows how much variation is accounted for for each factor   </li>
                <li><a href="images/eigenimages.jpg">eigenimages</a> 
                   <br />&nbsp; Qualitatively, eigenimages are the systematic variations of the images.
                   <br />&nbsp; Think of expressing each image as the average plus some linear 
                                combination of the eigenimages.                  </li>
                <li><a href="images/reconstituted.jpg">reconstituted images</a> 
                  -- these are characteristic images with extreme values for each factor</li>
              </ul>
            
            <p>To switch to PCA (or iterative PCA), change the register <i>x28</I> in
            <a href="ca-pca.msa">ca-pca.msa</a> to 2 (PCA) or 3 (iterative PCA). </p>
            
            <p>After running, examine the eigenimages and decide which ones to use. 
            Typically all but the first few are noisy. If not, increase the number 
            of eigenfactors to calculate, and re-run this PROCEDURE FILE.        </p>
        </ul>

        <center><p><TABLE BORDER="1">
            <caption align="bottom">
                The worm hemoglobin images are artificial and can be described with only a few factors. <br />
                The ribosome images are real, and require more factors to describe the data. <br />
                Note the vertical axis on the two histograms, representing the percent variation accounted for.
                </caption>
            <TR>
                    <TD><center><p align="center">
                        </p></center></TD>
                    <TD><center><p align="center">
                        factor 1 vs. 2<br />
                        <i>(click to enlarge)</I>
                        </p></center></TD>
                    <TD><center><p align="center">
                        eigenvalue histogram<br />
                        <i>(click to enlarge)</I>
                        </p></center></TD>
                    <TD><center><p align="center">
                        eigenimages<br />
                        <i>(click to enlarge)</I>
                        </p></center></TD>
                    <TD><center><p align="center">
                        reconstituted images<br />
                        <i>(click to enlarge)</I>
                        </p></center></TD>
                </TR>
            <TR>
                    <TD><center><p align="center">
                        worm <br />
                        hemoglobin <br />
                        (artificial)
                        </p></center></TD>
                    <TD><center><p align="center">
                        <a href="images/hb-factormap.gif">
                        <IMG SRC="images/hb-factormap-small.gif" border="2"></a>
                        </p></center></TD>
                    <TD><center><p align="center">
                        <a href="images/hb-histogram.gif">
                        <IMG SRC="images/hb-histogram-small.gif" border="2"></a>
                        </p></center></TD>
                    <TD><center><p align="center">
                        <a href="images/eigenimages.jpg"">
                        <IMG SRC="images/eigenimages.jpg" border="2" width="119" height="128"></a>
                        </p></center></TD>
                    <TD><center><p align="center">
                        <a href="images/reconstituted.jpg"">
                        <IMG SRC="images/reconstituted.jpg" border="2" width="167" height="128"></a>
                        </p></center></TD>
                </TR>
            <TR>
                    <TD><center><p align="center">
                        ribosome <br />
                        (real)
                        </p></center></TD>
                    <TD><center><p align="center">
                        <a href="images/factormap.gif">
                        <IMG SRC="images/factormap-small.gif" border="2"></a>
                        </p></center></TD>
                    <TD><center><p align="center">
                        <a href="images/histogram.gif">
                        <IMG SRC="images/histogram-sm32.gif" border="2"></a>
                        </p></center></TD>
                </TR>
        </TABLE></p></center>
    </ul>

    <li><b>Classification </b>-- choose one of three options: 

    <ol type="a">
        <li><p>K-means classification, using 
            <a href="../../man/clkm.html">'CL KM'</a><br />
            
            The advantage of K-means is that it is simple, 
            requiring primarily the number of classes to divide the particles into. 
            The simplicity is also the disadvantage; 
            there is no relation between the classes generated, 
            in contrast to the hierarchical methods below.</p>
            
        <ul>
            <li>PROCEDURE FILE: <a href="kmeans.msa">kmeans.msa</a>    </li>
            <li>INPUT PARAMETERS: number of factors, number of classes </li>
            <li>OUTPUT: class averages 

            <p>It can be informative to look at the individual particles from a class.  
                You can use 
                <a href="../../../../web/docs/web.html">WEB</a>/
                <a href="../../../../web/docs/jweb.html">JWEB</a>, or 
                <a href="../../../spire/tools-docs/tools.html">montagefromdoc</a>.  
                <br />
                Usage: <br />
                <code><b>montagefromdoc &nbsp; KMeans/docclass001.dat</B></code><br />
                If you have requested too many classes, there will be 
                 similar-looking class averages.  
                If you have requested too few, there will be dissimilar 
                particles within a class.  </p></li>
        </ul>
        
        <li>[Recommended] Ward's method, using 
        <a href="../../man/clhc.html">'CL HC'</a>  -- 
        The advantage is that, unlike Diday's method below, the 
        dendrogram branches to any desired number of classes, down in size to 
        individual particles. The disadvantage is that the dendrogram is unreadable if 
        there are so many branches. You can truncate the dendrogram in WEB/JWEB 
        as described below. 
            <ul>
                <li>PROCEDURE FILE: <a href="hierarchical.msa">hierarchical.msa</a> 
                <ul>
                    <li>INPUT PARAMETER: number of eigenfactors to use</li>
                    <li>OUTPUT: dendrograms (<a href="images/hcps.gif">PostScript</a> and 
                        SPIDER formats)
 
                    <p>After running, decide how many classes to use. 
                       The PostScript file may be highly branched, and nodes may be 
                        <a href="images/hcps.gif">unreadable</a>. 

                    <center><p><table border="1">
                            <tr>
                                <td><center><p align="center">
                                    <a href="images/hcps.gif">
                                    <img src="images/hcps-small.gif"> <br />
                                    Untruncated dendrogram</a>        <br />
                                    <i>(click to enlarge bottom row)</i>
                                </p></center></td>
                            </tr>
                    </table></p></center>

                    The SPIDER-format dendrogram document can be viewed 
                        with WEB/JWEB and 
                        truncated. In <a href="../../../../web/docs/web.html">WEB</a>, 
                        go to <i>Commands -&gt; Dendrogram </i> 
                        (<a href="images/hcweb5.gif"><i>example</i></a>).
 
                        In <a href="../../../../web/docs/jweb.html">JWEB</a>,
                        go to <i>File -&gt; Open SPIDER Document File</i>. 


        <center><p><table border="1">
                <tr>
                    <td><center><p align="center">
                        <a href="images/hcweb5.gif">
                        <img src="images/hcweb-small.gif"><br />
                        Dendrogram in X-Window 
                        <a href="../../../../web/docs/web.html">WEB</a></a><br />
                        <i>(click to enlarge)</i>
                    </p></center></td>
                </tr>
        </table></p></center></p>
                </ul>

    <li><a name="tree"><p>Visualize binary tree</a><br />
        It is often not clear where to truncate the dendrogram. 
        In X-Window <a href="../../../../web/docs/web.html">WEB</a>, 
          one only sees the terminal nodes in the dendrogram averaged.  
        (In <a href="../../../../web/docs/jweb.html">JWEB</a>, 
         averaged images in not implemented at the time of this writing, 
        although Bill Rice says that if the prefix is two characters long, it works.)  </p>
        
        <ul>
            <li>PROCEDURE FILE:  <a href="binarytree.msa">binarytree.msa</a>
                <ul>
                    <li>SUBROUTINES: <a href="averagenode.msa">averagenode.msa</a>, 
                                     <a href="update_lut.msa">update_lut.msa</a>  </li>

                    <li>INPUT PARAMETER: tree depth (number of averages will 
                           be (2**depth - 1))                                     </li>

                    <li>INPUTS: dendrogram (from 
                        <a href="hierarchical.msa">hierarchical.msa</a>), images  </li>

                    <li>OUTPUTS: averages (unlabeled and labeled)                 </li>
                </ul>

            <li>Visualize the output using <a href="../../../spire/tools-docs/tools.html">binarytree.py</a> 
                Syntax:                                                                               <br />
                <i>binarytree &nbsp;&nbsp; labeled001.dat &nbsp;&nbsp; 4 &nbsp;&nbsp; 2 &nbsp;&nbsp; 1024</i> <br />
                where:
                <ul>
                    <li><b>labeled001.dat</b> is an example filename (without a wild card)            </li>
                    <li><i>4 (optional)</i> is the tree depth, i.e., the number of rows (default is 6)</li>
                    <li><i>2 (optional)</i> is the margin width (default is 2)                        </li>
                    <li><i>1024 (optional)</i> is the canvas width                                    </li>
                </ul>
            <li>If <a href="../../../spire/tools-docs/tools.html">SPIDER's python tool -- binarytree.py</a> 
                is not installed, try <a href="spidertree.msa">spidertree.msa</a>. 
                The output is a SPIDER-format image. However, the file size
                may be very large. 
                <ul> 
                    <li>INPUT PARAMETERS: tree depth (number of averages will be (2**depth - 1))</li>

                    <li>INPUTS: averages from <a href="binarytree.msa">binarytree.msa</a>       </li>

                    <li>OUTPUTS: SPIDER-format tree image                                       </li>
                </ul> 

            <center><p><table border="1">
                <tr>
                   <td><center><p align="center">
                       <a href="images/tree3.jpg">
                       <img src="images/tree3-small.jpg">   <br />
                       binarytree.py, <i>depth=4</i></a>    <br />
                       <i>(click to enlarge)</i>

                   <td><center><p align="center">
                       <a href="images/tree4.jpg">
                       <img src="images/tree4-small.jpg">  <br />
                       tree.msa, <i>depth=4</i></a>        <br />
                       <i>(click to enlarge)</i>
                       </p></center></td>
                </tr>
            </table></p></center>
        </ul>
        
        </ul>
</ul>
        <p />

        <li>Diday's method, using <a href="../../man/clcla.html">'CL CLA'</a> -- 
            I hear that this method  works exceedingly well. In 
            practice though, I find that I have limited control over the number of 
            classes, which may or may not be a problem depending on the 
            application. Also, I sometimes get errors with large data sets with this method. 
        <ul>
            <li>PROCEDURE FILE: <a href="cluster.msa">cluster.msa</a> 
            <ul>
                <li>INPUT PARAMETER: number of eigenfactors to use </li>
                <li>OUTPUT: dendrograms 
                    (<a href="images/dendrogram.gif">PostScript</a>
                    and SPIDER formats)
                <p>After running, decide how many classes to include. using 
                   <a href="../../../../web/docs/web.html">WEB</a>/
                   <a href="../../../../web/docs/jweb.html">JWEB</a> 
                (<i>Commands -&gt; Dendrogram</i>) and clicking on 
                 <i>Show averaged images</i>.</p></li> 
            </ul>

            <center><p><table border="1">
                <tr>
                    <td><center><p align="center">
                        <a href="images/dendrogram.gif">
                        <img src="images/dendrogram-sm4.gif"> <br />
                        dendrogram                            <br />

                        PostScript format</a>                 <br />
                        <i>(click to enlarge)</i>
                    </p></center></td>
                </tr>
            </table></p></center>

            <li>PROCEDURE FILE: <a href="classavg.msa">classavg.msa</a> 
                <ul>
                   <li>INPUT PARAMETER: desired number of classes </li> 
                   <li>OUTPUT: class averages                     </li> 
                </ul>
        </ul>
        <p />

    </ol>
</ol>

<hr />

<p><b><font size="4">Full version</font></b></p>

<p>
This full version is the 2D alignment scheme which I use routinely. 
It is a subset of the steps for alignment and classification for 
untilted-specimen images for random conical tilt.
</p>

<ul>

<li><p><b>[Recommended] Generate parameter file</b></p>           </li>
    
    <p>
    If using <font class="guitool">SPIRE</font>, a parameter doc file will be generated 
    upon starting a project. While not strictly required, without this file, 
    you will need to remember to check for these frequently-used parameters each time.  </p>
    
    <p>If not using <font class="guitool">SPIRE</font>, run <a href="makeparams.msa">makeparams.msa</a>, 
    which will generate <font class="output">params</font>, a doc file containing the 
    <a href="params.html">reconstruction parameters</a>.
    </p>

    Not all 20 parameters are used, so here are the key values:
    
    <table class="outs">
    <tr valign="top"><td>&curren;                  </td>
        <td><b><i>5</i></b>:                       </td>
        <td>pixel size, in Angstroms.</td></tr>
    <tr valign="top"><td>&curren;                  </td>
        <td><b><i>17</i></b>:                      </td>
        <td>window size, in pixels.</td></tr>
    <tr valign="top"><td>&curren;                  </td>
        <td><b><i>18</i></b>:                      </td>
        <td>particle diameter, in pixels.          </td></tr>
    </table>
    
    <p>Alternatively, you can copy and modify this file from elsewhere. 
       An example file called <font class="input">bak-params</font> is present in the tarball.</p>
    

<li><p><b>Prepare particle stack.</b></p></li>
    
    <p>The following procedures will expect particle images in the form of a single stack, 
       plus a doc file with a list of particles.
       If your images are not in this form, we present two methods to generate them:
    </p>
    
    <ol type-a>
        <li><p><b>Combine unstacked images.</b></p> </li>
        
            <p> By default, the procedure <a href="win2stack.msa">win2stack.msa</a> 
                assumes that the input unstacked images are of the form 
                <font class="input">Particles/win*****</font>. 
                Edit as necessary. 
            </p>
            
            <p> Also, enter the highest particle number that you wish to import. 
                The procedure will look for the existence of particles up until this number.
            </p>
        
        <li><p><b>Concatenate a set of sets into one stack.</b></p> </li>
        
            <p> The procedure <a href="combinestacks.msa">combinestacks.msa</a> 
                will look for the presence of a series of stack files, 
                such as <font class="input">Particles/winser_*****</font>. 
                Programs such as <font class="guitool">e2boxer.py</font>, for example, 
                may generate a stack of images per micrograph.
            </p>
            
            <p> Also, enter the stack particle number that you wish to import. 
                The procedure will look for the existence of stacks up until this number.
            </p>
    </OL>
    
    <b>Outputs, for either of the two cases above:</b><br />

    <table class="outs">
        <tr valign="top"><td>&curren;                      </td>
            <td><font class="output">listparticles</font>: </td>
            <td>A list of windowed particles.              </td></tr>
        <tr valign="top"><td>&curren;                      </td>
            <td><font class="output">stk2-unaligned</font>:</td>
            <td>Single particle stack.                     </td></tr>
        <tr valign="top"><td>&curren;                      </td>
            <td><font class="output">stk1-unaligned_backup</font>:</td>
            <td>Generated in case high-pass filtration needs to be repeated.</td></tr>
    </table>
    
    </p>
    
<li><p><b>[Recommended for negatively-stained images] High-pass filter windowed images</b></p>
    
    <p>For negatively-stained images, it is often the case that the puddle of stain surrounding the particles 
    dominates the classification (although maybe not the case for deep-stained specimens).  
    In this case, it may be helpful to high-pass filter the images using a filter radius 
    larger than the size of your particle.  If you cut off <u>too</u> much of the low-frequency information 
    however, the aligning power of the images will be lost.  You may need to tweak the radii.</p> 
    
    <p>If you are unsure whether to do these steps, feel free to skip them, 
    but be sure to look at the reconstituted images when you run <a href="ca-pca.msa">ca-pca.msa</a> 
    in order to see if the stain is the predominant source of variation. </p> 
    
    <p>The PROCEDURE FILE <a href="highpass.msa">highpass.msa</a> reads the stack
    <font class="input">stk1-unaligned_backup</font> and generates:</p>
    
    <table class="outs">
        <tr valign="top"><td>&curren;</td>
            <td><font class="output">stk2-unaligned</font>:</td>
            <td>High-pass filtered untilted-specimen image stack (overwrites previous version).</td></tr>
    </table>

<li><p><b>Align the untilted-specimen particles using reference-free alignment.</b></p></li>

    <p> Before running classification, the particles will first be run through reference-free alignment.  
    Even if there are different views of particles (orientation or conformation), 
    the idea is that these different classes of particles will align consistently with 
    the type of particle to which it is aligned.
    </p>
    
    <p> There are a few different implementations of reference-free alignment, such as 
    <a href="../../man/apsr.html">'AP SR'</a>. 
    The procedure file used here, <a href="pairwise.msa">pairwise.msa</a>, uses an algorithm called 
    the "pyramidal system for prealignment construction."  For more information, see the quick version above.  
    </p>
    
    <p> Briefly, <a href="pairwise.msa">pairwise.msa</a> takes the first two images from the stack
    <font class="input">stk2-winrot</font>, aligns them, and averages them.  
    In the next iteration, it will align pairs of averages, and then average them,  and so forth.  
    This type of alignment appears to be less random than does 
    <a href="../../man/apsr.html">'AP SR'</a>, 
    which chooses seed images as alignment references.</p>

    <b>Outputs:</b><br />
    
    <table class="outs">
            <tr valign="top"><td>&curren;</td>
                <td><font class="output">Pairwise/stkcentavg</font>:</td>
                <td>Stack of centered intermediate averages, with circles overlain.
                    <br />See note about centering on the 
                    <a href="align.html">alignment parameters page</a>.</td></tr>
            <tr valign="top"><td>&curren;</td>
                <td><font class="output">stk3-aligned</font>:</td>
                <td>Aligned untilted-specimen image stack.
                    <br />This stack will be overwritten with subsequent alignments.</td></tr>
            <tr valign="top"><td>&curren;                            </td>
                <td><font class="output">Pairwise/stkaligned</font>: </td>
                <td>Copy of aligned untilted-specimen image stack,
                    <br />retained locally to the <font class="output">Pairwise/</font> directory.</td></tr>
            <tr valign="top"><td>&curren;</td>
                <td><font class="output">Pairwise/rfreeavg001</font>:</td>
                <td>Final average of all images.                     </td></tr>
            <tr valign="top"><td>&curren;                            </td>
                <td><font class="output">Pairwise/docalign</font>:   </td>
                <td>Alignment doc file.                              </td></tr>
    </table>

<li><p><b>Low-pass filter and shrink the aligned particles.</b>      </p></li>

    The procedure file <a href="filtershrink.msa">filtershrink.msa</a> 
    low-pass filters and optionally shrinks the particle images. 
    The next step, multivariate data analysis, will be faster if 
    the image stack <font class="input">stk3-aligned</font> are made smaller. 
    Also, the individual images will be examined subsequently and 
    will be easier to recognize if they are low-pass filtered.</p>

    <b>Outputs:</b><br />
    
    <table class="outs">
            <tr valign="top"><td>&curren;                   </td>
                <td><font class="output">stkfiltered</font>:</td>
                <td>Low-pass, shrunken filtered image stack.</td></tr>
    </table>
    
    <p> The default parameters are hopefully reasonable. 
        To view the output filtered images, type:
        <br /><code class="snippet">montagefromdoc listparticles.spi stkfiltered.spi</code></p>
        
    <p> Adjust the filtration parameters as necessary, and re-run. </p>

<li><p><b>Run multivariate statistical analysis (MSA) on filtered images.</b></p></li>

    <p> The procedure file <a href="ca-pca.msa">ca-pca.msa</a> runs 
    correspondence analysis (CA) or principal component analysis (PCA) or iterative PCA (IPCA). 
    In principle, one can classify the image stack <font class="input">stkfiltered</font> 
    without running MSA, but here we will be using it. </p>
    
    <p> What this means, briefly, is that systematic variations are reduced into 
    an arbitrary number of factors (<i>e.g.,</I> 25 used here); 
    in this case, the factors can be expressed as images, or "eigenimages." 
    Each image can be reconstituted as the sum of these eigenimages, when using the proper weights.  
    How CA and PCA differ is in the way these weights are calculated. 
    The "importance" of each factor is the percent variation that is accounted for. 
    The eigenimages of lower importance typically correspond to noise. 
    Thus, an image reconstituted from the strongest eigenimages can be thought of as 
    a type of filtered image, where some contribution of noise has been excluded.
    </p>
    
    <p> For a more complete description of multivariate statistical analysis 
    (or more accurately, multivariate data analysis), see Chapter 4 in 
    <a href="#jf2006">Frank, 2006</a>.
    For details about the specific SPIDER commands pertaining to MSA, and example outputs, see the
    <a href="../classification/tutorial.html">classification and clustering tutorial</a>.</p>
    
       <b>Outputs:</b><br />
       
       <table class="outs">
        <tr valign="top"><td>&curren;</td>
            <td><font class="output">MSA/eigenimg</font>:  </td>
            <td>Stack of eigenimages, i.e., images showing systematic correlated variations between pixels.
                <br />For more information about eigenimages (and reconstituted images), see the
                <a href="../classification/tutorial.html#virtual">
                classification and clustering tutorial</a>.</td></tr>
        <tr valign="top"><td>&curren;</td>
            <td><font class="output">MSA/reconst</font>:   </td>
            <td>Stack of images reconstituted using a single factor
                <br />Factors can be positive or negative, hence the two versions for each.
                <br />For more information about reconstituted images (and eigenimages), see the
                <a href="../classification/tutorial.html#virtual">
                classification and clustering tutorial</a>.</td></tr>
    <tr valign="top"><td>&curren;</td>
        <td><font class="output">MSA/ps_eigenvalues.ps</font>:</td>
        <td>A PostScript file with a histogram of the percent variation for each factor.</td></tr>
    <tr valign="top"><td>&curren;</td>
        <td><font class="output">MSA/cas_IMC</font>:          </td>
        <td>A text file containing information about each factor and for each image
        <br />For more information about the output of
        <a href="../../man/cas.html">'CA S'</a>, see the
        <a href="../classification/tutorial.html#CAPCA">
        classification and clustering tutorial</a>.           </td></tr>
    <tr valign="top"><td>&curren;</td>
        <td><font class="output">MSA/factormapAABB</font>:    </td>
        <td>PostScript format plots for factor #AA <i>vs.</i> factor #BB (where BB = AA+1)
        <br />Clusters are indicative of distinct classes.    </td></tr>
    </table>

    <br /><i>Important parameters:</I>
    
    <table class="outs">
        <tr valign="top"><td>&curren;                                   </td>
        <td><b><i>[cas-option]</I></b>:                                 </td>
        <td>Option of CA (option 1), PCA (2), or iterative PCA (3)
            <br />One method or another sometimes crashes due to "numerical inaccuracy."
            I subjectively try CA first, then iterative PCA, then PCA last.</td></tr>
        <tr valign="top"><td>&curren;                                   </td>
        <td><b><i>[num-factors]</I></b>:                                </td>
        <td>Number of factors to calculate.
            <br />If after you run this procedure file, all of the eigenimages look meaningful,
            then increase this parameter and re-run.                    </td></tr>
    </table>
    
    <p> <u>NOTE</u>: For negatively-stained samples, when looking at the reconstituted images, 
        it may be that the stain distribution is the greatest source of variation. 
        If this is the case, go back to the high-pass filtration step above.
    </p>

<li><p><b>Classify images.</b></p></li>

    <p> At this step, we willl attempt to separate the images into homogeneous subclasses.  
    There is a tradeoff between homogeneity of the subclasses, and 
    size of the subclasses (which is related to the improvement of signal-to-noise in the class averages). 
    In other words, if too few classes are used, dissimilar classes will be grouped together, and
    if too many classes are used, class averages will be likely redundant and noisy.
    Thus, there is some degree of subjectivity involved in the classification.
    </p>
    
    <p> For any of the classification options presented here, 
    choose the number of factors to use (parameter <i><b>[numFactors]</b></I>) 
    based on the appearance of the eigenimages and the strength of the factors (based on the histogram), 
    including those factors that you believe to represent true structural differences. 
    Use of weaker factors will probably not make a appreciable difference, however.
    </p>
    
    <p> There are three classification options given here. 
    (For hints about the usage of the different methods, see the quick version above.)
    </p>
    
    <p> To view the particles belonging to each class, refer below to the 
    <a href="#vbv">instructions about <font class="guitool">verifybyview</font></a>.</p>
    
    <br />Choose one of the following 3 classification options:
      <ol type-a>
        <li><b>[Recommended] Hierarchical classification, using Ward's method.</b> </li>
          <ol>
        <li><p><b>Run hierarchical classification.</b></p> </li>
        
            <p> This classification method uses SPIDER command 
                 <a href="../../man/clhc.html">'CL HC'</a>. 
            For a description of the outputs, see the 
            <a href="../classification/tutorial.html#clhc">
            classification and clustering tutorial</a>.
            </p>
            
            <p>The procedure file <a href="hierarchical.msa">hierarchical.msa</a> 
            reads the MSA factor text file <font class="input">MSA/cas_IMC</font>.</p>
        
            <b>Output:</b><br />
            
            <table class="outs">
            <tr valign="top"><td>&curren;</td>
                <td><font class="output">Hierarchical/docdendro</font>:</td>
                <td>Dendrogram doc file, showing the relatedness between images.
                <br />See the <a href="../../../techs/classification/tutorial.html#clhc">
                classification and clustering tutorial</a> for more details.
                </td></tr>
            </table>
            
        <li><b>Generate class averages.</b></li>
        
        <p> In the previous step, we didn't explicitly assign classes. 
            We simply made the dendrogram, which describes the relatedness of the images. 
            In this step, we will explicitly pare down the dendrogram into classes.</p>
            
        Choose one of:      
          <ol type-a>
            <li><b>[Recommended] Generate binary tree</b></li>
            
            <p> It is somewhat arbitrary where to cut off the dendrogram 
                <font class="input">Hierarchical/docdendro</font> generated above. 
                This procedure file will generate class averages for <u>all</u> depths until a specified cutoff, 
                averaging the image stack <font class="input">stkfiltered</font>.
                The output class averages are arranged into the format of a binary tree, such that 
                the average is the top-level node, the first two subclasses are at the next depth, 
                the first two classes of each of those two classes are at the next depth, etc.
                While it becomes straightforward how to draw the tree, the drawback is that 
                information is lost about the height of the branches.  
                That is, the height of the branches in the binary tree does not direct correspond to 
                the actual height of the branch in the dendrogram.
            </p>
                
            <p> In the procedure file <a href="binarytree.msa">binarytree.msa</a> 
                (which calls <a href="averagenode.msa">averagenode.msa</a> 
                and <a href="update_lut.msa">update_lut.msa</a>), 
                the parameter <i><b>[end-depth]</b></i> 
                will call for 2**<i><b>[end-depth]</b></I> class averages. 
                It is better to err on the side of too high a depth and display fewer in the next step.
            </p>
                
                <b>Outputs:</b><br />
            
                <table class="outs">
                                <tr valign="top"><td>&curren;</td>
                                    <td><font class="output">Tree/classavg***</font>:</td>
                                    <td>Class average for each node on the binary tree.</td></tr>
                                <tr valign="top"><td>&curren;</td>
                                    <td><font class="output">Tree/labeled***</font>:</td>
                                    <td>Class average, labeled with the class number and the number of particles.</td></tr>
                </table>
                
            <p><u>NOTE</u>: The format of SPIDER's dendrogram doc file changed in version 17.13. 
                Previously the top two branches had height 1, whereas now, it's 100. 
                If this procedure file crashes, check the parameter <b><i>[top-branch]</I></b>.
            </p>
                
            <p> Instructions for viewing the binary tree are given below, 
                in the section below, on the <a href="#treepy">selection of class averages</a>.
            </p>
            
            <li><b>Generate a pre-specified number of class averages</b></li>
            
            <p> To use this option, run procedure file <a href="classavg.msa">classavg.msa</a>. 
                        This option cuts off the dendrogram <font class="input">dendrodoc</font> at a specific height.  
            The height determines the number of classes into which the dendrogram is pared. 
            The number of classes desired is specified by the parameter <i><b>[numClasses]</b></i> in 
            <a href="classavg.msa">classavg.msa</a>. 
            However, because of the imprecision in how the appropriate height in the dendrogram is searched, 
            the actual number of classes found may not exactly match the desired number of classes. 
            </p>
            
            <b>Outputs:</b><br />
            <table class="outs">
                            <tr valign="top"><td>&curren;</td>
                                <td><font class="output">Hierarchical/classavg***</font>:</td>
                                <td>Class averages.</td></tr>
                            <tr valign="top"><td>&curren;</td>
                                <td><font class="output">Hierarchical/docclass***</font>:</td>
                                <td>List of particles belonging to each class.</td></tr>
                            <tr valign="top"><td>&curren;</td>
                                <td><font class="output">Hierarchical/listclasses</font>:</td>
                                <td>Doc file with each information about each class <br />
                                   Needed as an input for <font class="guitool">verifybyview.py</font> 
                                  (<a href="#vbv">more information below</a>).</td></tr>
                            <tr valign="top"><td>&curren;</td>
                                <td><font class="output">Hierarchical/classvar***</font>:</td>
                                <td>Class variances.</td></tr>
            </table>
          </ol>
          </ol>

        <li><p><b>Run K-means classification</b></p> </li>
        
        <p> K-means classification uses the SPIDER command
        <a href="../../man/clkm.html">
        'CL KM'</a>.  In contrast to the other classification options presented here, 
        K-means will generate exactly the number of classes requested.  
        The procedure file <a href="kmeans.msa">kmeans.msa</a> reads 
        <font class="input">MSA/cas_IMC</font> and, in a single step, 
        generates <i>K</I> (parameter <i><b>[numClasses]</I></b> in the procedure file) classes 
        from the image stack <font class="input">stkfiltered</font>.
        </p>
        
        <b>Outputs:</b><br />
            <table class="outs">
                        <tr valign="top"><td>&curren;</td>
                            <td><font class="output">Kmeans/docassign</font>:</td>
                            <td>Doc file of analogous format to 
                                dendrograms from the other 2 classification methods.</td></tr>
                        <tr valign="top"><td>&curren;</td>
                            <td><font class="output">Kmeans/classavg***</font>:</td>
                            <td>Class averages.</td></tr>
                        <tr valign="top"><td>&curren;</td>
                            <td><font class="output">Kmeans/docclass***</font>:</td>
                            <td>List of particles belonging to each class.</td></tr>
                        <tr valign="top"><td>&curren;</td>
                            <td><font class="output">Kmeans/listclasses</font>:</td>
                            <td>Doc file with each information about each class
                                <br />Needed as an input for <font class="guitool">verifybyview.py</font> 
                                (<a href="#vbv">more information below</a>).</td></tr>
                        <tr valign="top"><td>&curren;</td>
                            <td><font class="output">Kmeans/classvar***</font>:</td>
                            <td>Class variances.</td></tr>
            </table>
        
        <li><p><b>Clustering using Diday's method</b></p> </li>
          <ol>
        <li><p><b>Run hierarchical classification.</b></p> </li>
        
            <p> The procedure file <a href="cluster.msa">cluster.msa</a> uses SPIDER command
            <a href="../../man/clcla.html">
            'CL CLA'</a>, which uses Diday's method of moving centers. 
            Like with Ward's method above, this method is hierarchical, 
            reading <font class="input">MSA/cas_IMC</font>.</p>
            
                <b>Outputs:</b><br />
                
                <table class="outs">
                <tr valign="top"
                    <td>&curren;</td>
                    <td><font class="output">Cluster/docdendro</font>:</td>
                    <td>Dendrogram doc file, showing the relatedness between images. <br />
                         See the <a href="../classification/tutorial.html#clhc">
                         classification and clustering tutorial</a> for more details.</td>
                </tr>
                </table>
            
        <li><b>Generate class averages.</b></li>
        
            <p> With the previous step, analogously to Ward's method above, 
            we didn't explicitly assign classes. 
            We simply made the dendrogram 
            <font class="input">Cluster/docdendro</font>, 
            which describes the relatedness of the images. </p>
            
            <p> In the procedure file <a href="classavg.msa">classavg.msa</a>, 
            we will explicitly pare down the dendrogram into classes. 
            (I haven't been able to make Diday's method compatible with 
            <a href="binarytree.msa">binarytree.msa</a>,
            so for now, it's functionally equivalent to K-means.) </p>
            
            <p> <u>NOTE:</u> The default parameter <i><b>[class-type]</b></I> 
                        in the procedure file is for Ward's hierarchical method, 
            so change that parameter to <i><b>2</b></I>.
            </p>
            
            <b>Outputs:</b><br />
                <table class="outs">
                                <tr valign="top"><td>&curren;</td>
                                    <td><font class="output">Cluster/classavg***</font>:</td>
                                    <td>Class averages.</td></tr>
                                <tr valign="top"><td>&curren;</td>
                                    <td><font class="output">Cluster/docclass***</font>:</td>
                                    <td>List of particles belonging to each class.</td></tr>
                                <tr valign="top"><td>&curren;</td>
                                    <td><font class="output">Cluster/listclasses</font>:</td>
                                    <td>Doc file with each information about each class
                                        <br />Needed as an input for <font class="guitool">verifybyview.py</font> 
                                        (<a href="#vbv">more information below</a>).</td></tr>
                                <tr valign="top"><td>&curren;</td>
                                    <td><font class="output">Cluster/classvar***</font>:</td>
                                    <td>Class variances.</td></tr>
                </table>
          </ol>
      </ol>
      
    <p> It may be informative to see the individual particles that belong to each class. 
    This can be accomplished by using 
    <a href="../../../spire/tools-docs/verifybyview.html"> <font class="guitool">verifybyview.py</font></a>. 

    There should be a settings file, <a href=".verifybyview">.verifybyview</a>, 
    present in the tarball. For more details, refer below to the 
    <a href="#vbv">section on viewing the class averages and constituent particles</a>.
    </p>    

<li><p><b>Select unique classes</b></p></li>

    <p> In the next step, we will select class averages to use as references for multireference alignment. 
    The first option applies to the output of <a href="binarytree.msa">binarytree.msa</a> above. 
    The second option applies to all classification options from above, 
    including the output of <a href="binarytree.msa">binarytree.msa</a>.
    </p>
    
      <ol type-a>
    <li><a name="treepy"><b>View and select classes from binary tree</b></a></li>
    
        <p> To view the binary tree, <code class="snippet">cd Tree</code> and run:
        <br /><code class="snippet">binarytree labeled001.spi 4 goodclasses.spi</code>
        </p>
        
        <p> The parameter <b><i>4</I></b> on the command line above specifies 
        the maximum depth displayed of the tree.  
        A depth of <b><i>2</I></b>, in comparison, 
        would display only the combined average plus 
        the two classes corresponding to the first branch in the dendrogram. 
        Unlike the <i>Dendrogram</I> option in 
        <a href="http:../../../../web/docs/web.html"><font class="guitool">WEB</font></a>, 
        one advantage of <font class="guitool">binarytree.py</font> is that 
        averages are also shown for the non-terminal branches. 
        A disadvantage is that the height of the branches in the tree are purely relative, 
        and have no bearing to the numerical height of the branch point in the dendrogram
        (and thus the relatedness of the classes).
        </p>

        <p> As one proceeds down the tree, the number of classes increases, and 
        the number of particles per class decreases. 
        As is the case with classification in general, 
        there is a tradeoff between too big a class, 
        such that dissimilar particles are combined, and 
        too small a class, where the signal-to-noise ratio is low. 
        One strategy for displaying an appropriate depth is 
        the point at which a parent class's subclasses do not look different than the parent class. 
        </p>

        <p> Save the selection file with <i>CTRL-s</i>. 
        Further options for the program are described on the <a href="index.html"> MSA page</a>.
        </p>
        
        <p>The script <a href="../../../spire/tools-docs/tools.html">
                      <font class="guitool">binarytree.py</font></a>                 
        requires that you have <a href="../../../spire/install.html"> 
        installed SPIDER's python tools</a>.
        
        If you don't have <font class="guitool">SPIDER's python tools</font></a> installed, 
        you can alternatively run <a href="binarytree.msa">tree.msa</a>, 
        which generates a (large) SPIDER image, <font class="output">Tree/tree*</font>. 
        To generate a selection file of class averages, <code class="snippet">cd Tree</code>, 
        and either categorize the montage of class averages in 
        <font class="guitool">WEB</font> or <font class="guitool">JWEB</font>, 
        or manually create the selection file interactively in SPIDER using 
        <a href="../../man/doccreate.html">
        'DOC CREATE'</a>.  The selection file should be called 
        <font class="output">Tree/goodclasses</font>.
        </p>
        
    <li><b><a name="vbv">View the class averages and, optionally, the constituent particles using 
        <font class="guitool">verifybyview.py</font></a></b></li>
        
        <p> This option can be used for any of the classification schemes described above. 
        It can also be used in conjunction with <font class="guitool">binarytree.py</font> to 
        view the individual particles that belong to a class.
        </p>
        
        <p> Open <font class="guitool">verifybyview.py</font> by typing: <br />
        <code class="snippet">verifybyview Tree</code>    &nbsp; if you used the binary-tree option, or <br />
        <code class="snippet">verifybyview Kmeans</code>  &nbsp; if you ran K-means classification, or  <br />
        <code class="snippet">verifybyview Cluster</code> &nbsp; if you used Diday's clustering method, <i>etc.</I>
        </p>

               
        <br />Included in the tarball is a starting <a href=".verifybyview">.verifybyview</a> file 
        with mostly reasonable settings. 
        The primary purpose of that <a href=".verifybyview">.verifybyview</a> file, however, 
        was to view classes for multireference alignment, so 
        you may wish to change the following 2 settings:
        
        <table class="outs">
            <tr valign="top"><td>&curren;                                        </td>
            <td><i>Display &gt; <b>reverse order</b></i>:                        </td>
            <td>Toggling this option will display the averages starting from #1. </td></tr>
            <tr valign="top"><td>&curren;                                        </td>
            <td><i><i>Display &gt; labels &gt; <b>text label</b></i></i>:        </td>
            <td>Change <i><b>CCC</b></I> to <i><b>N</b></i>
                for the number of particles in a class.</td></tr>
        </table>

<!--            <p> The <a href=".verifybyview">.verifybyview</a> file is updated with each change. 
        If you make a mistake, or wish to revert to the original, 
        there are copies the of the file included in the tarball, called
        <a href=".verifybyview.tree">.verifybyview.tree</a> and 
        <a href=".verifybyview.multiref">.verifybyview.multiref</a>.
        </p>-->
        
        If <a href=".verifybyview">.verifybyview</a> is not present 
        (it is a hidden file, starting with a period, so to list it, type 
        <code class="snippet">ls -a</code> in your toplevel directory), 
        you will need to adjust the filenames. 
        In the initial popup window, the values should be:
        
        <table class="outs">
                    <tr valign="top"><td>&curren;</td>
                        <td><font class="input">.spi</font>:</td>
                        <td>Data extension. Adjust as necessary</td></tr>
                    <tr valign="top"><td>&curren;</td>
                        <td><font class="input">Tree</font>:</td>
                        <td>Initial directory.</td></tr>
                    <tr valign="top"><td>&curren;</td>
                        <td><font class="input">listclasses</font>:</td>
                        <td>List of classes.</td></tr>
                    <tr valign="top"><td>&curren;</td>
                        <td><font class="input">classavg***</font>:</td>
                        <td>Class-average template.</td></tr>
        </table>

        <br />The first time you click on a class average, 
        a popup menu will appear to confirm filenames, 
        some of which are repeated from the initial popup menu. 
        The key filenames are:
        
        <table class="outs">
                    <tr valign="top"><td>&curren;</td>
                        <td><font class="input">docclass***.spi</font>:</td>
                        <td>List of particles belonging to each class.</td></tr>
                    <tr valign="top"><td>&curren;</td>
                        <td><font class="input">Filtered/flt*****.spi</font>:</td>
                        <td>Filtered-image template.</td></tr>
                    <tr valign="top"><td>&curren;</td>
                        <td><font class="output">goodclasses.spi</font>:</td>
                        <td>List of selected classes.</td></tr>
        </table>
        
        <p> When you click on a class average 
        (except for the first instance, when a popup menu appears), 
        a montage of individual particles will appear. 
        <br /><u>NOTE</u>: There is a limit of how many images that can be displayed in one montage. 
        Clicking on a class with ~1000 particles will show an error message to the console window.
        </p>

        <p> It is unnecessary to verify individual particles at this time. 
        Instead, simply select the class averages that you wish to use for multireference alignment. 
        Select the classes averages by right-clicking on them. 
        When you are finished, save the selection by going to the 
        <i>File</I> menu and selecting <i>Save selection</I> (or <i>ALT-s</I>).
        </p>
      </ol>


<li><p><b>Run multireference alignment</b></p></li>

    <p>The procedure file <a href="multirefalign.msa">multirefalign.msa</a> 
    takes the class averages <font class="input">classavg***</font> 
    selected in the previous step in <font class="input">goodclasses</font>, 
    re-centers them, and uses them for multireference alignment. 
    The previously aligned image stack <font class="input">stk3-aligned</font> 
    is used to generate the reference images 
    (the class averages are derived from filtered images which may have been downsampled), 
    and the unaligned image stack <font class="input">stk2-winrot</font> 
    will be aligned to those references. 
    </p>
    
    <p><u>NOTE</u>: Ensure that the classification method that you used 
    is reflected in the procedure file label <b><i>[old_class_dir]</i></b>. 
    If you used K-means, for example, use the directory <font class="input">Kmeans</font>.
    The default in the procedure file is <font class="input">Tree</font>. 
    </p>
    
    <b>Outputs:</b><br />
    <table class="outs">
            <tr valign="top"><td>&curren;</td>
                <td><font class="output">Multiref1/stkref</font>:</td>
                <td>Stack of reference images used for alignment.
                    <br />They should resemble full-sized, unfiltered versions of the selected class averages</td></tr>
            <tr valign="top"><td>&curren;</td>
                <td><font class="output">Multiref1/docalign</font>:</td>
                <td>Alignment parameters
                    <br />See the documentation for 
                    <a href="../../man/apsh.html">
                    'AP SH'</a> for an explanation of the doc file.</td></tr>
            <tr valign="top"><td>&curren;</td>
                <td><font class="output">Multiref1/selview***</font>:</td>
                <td>A list of particles belonging to each class.</td></tr>
            <tr valign="top"><td>&curren;</td>
                <td><font class="output">Multiref1/stkcenu</font>:</td>
                <td>Aligned untilted-specimen image stack.
                    <br />This copy is retained locally to the <font class="output">Pairwise/</font> directory.</td></tr>
            <tr valign="top"><td>&curren;</td>
                <td><font class="output">stk3-aligned</font>:</td>
                <td>Aligned untilted-specimen image stack.
                    <br />This stack will be overwritten with subsequent alignments.</td></tr>
            <tr valign="top"><td>&curren;</td>
                <td><font class="output">Multiref1/viewavg***</font>:</td>
                <td>Average of particles belonging to each reference.</td></tr>
            <tr valign="top"><td>&curren;</td>
                <td><font class="output">docmultiref</font>:</td>
                <td>Keeps track of successfully executed iterations of multireference alignment.</td></tr>
    </table>


<li><p><b>Re-filter aligned images</b></p></li>

    <p> With newly aligned images, we will re-filter and optionally shrink the images before re-classifying them. 
    The newly aligned image stack, <font class="input">stk3-aligned</font>, 
    overwrote the old version, thus the inputs and outputs of <a href="filtershrink.msa">filtershrink.msa</a> 
    will remain the same.
    </p>

    <b>Outputs:</b><br />
    
    <table class="outs">
    <tr valign="top"><td>&curren;                   </td>
        <td><font class="output">stkfiltered</font>:</td>
        <td>Low-pass, shrunken filtered image stack.</td></tr>
    </table>
    
    <p> Adjust the parameters as necessary, 
    although the parameters used previously should be applicable.
    </p>

<li><p><b>Classify the particles assigned to each reference</b></p></li>

    <p> In the next step, we will perform separate classifications for the particles belonging to each reference. 
    In principle, the particles assigned to each reference should be identical. 
    However, if too few references were chosen, there may be homogeneous subsets assigned to a reference. 
    Also, if there are subtle but systematic structural differences, 
    those difference may be noticeable upon MSA and classification. 
    Also, junk particles (damaged particles or contaminants) may not match any references well. 
    </p>
    
    <p> The procedure file <a href="classifybyview.msa">classifybyview.msa</a> 
    performs separate classifications on the image stack <font class="input">stkfiltered</font>
    for each of the selection doc files <font class="input">Multiref1/selview***</font>. 
    As a rough measure of the quality of each particle and each class average, 
    the cross-correlation coefficient (CCC) is computed against the reference 
    <font class="input">Multiref1/ref***</font>.
    </p>
    
    <b>Outputs</b>:<br />
    
    <table class="outs">
            <tr valign="top"><td>&curren;</td>
                <td><font class="output">Multiref1/listclasses</font>:</td>
                <td>List of classes for all references
                    <br />Also contains various statistics for each class.</td></tr>
            <tr valign="top"><td>&curren;</td>
                <td><font class="output">Multiref1/classavg***</font>:</td>
                <td>Class averages.</td></tr>
            <tr valign="top"><td>&curren;</td>
                <td><font class="output">Multiref1/docclass***</font>:</td>
                <td>List of particles belonging to each class.        </td></tr>
    </table>

    <br /><i>Notable parameters include:</I>
    
    <table class="outs">
        <tr valign="top"><td>&curren;                                   </td>
        <td><b><i>[reduce-factor]</I></b>:                              </td>
        <td>Reduction factor applied in <a href="filtershrink.msa">filtershrink.msa</a>.</td></tr>
        <tr valign="top"><td>&curren;                                   </td>
        <td><b><i>[ca-pca]</I></b>:                      </td>
        <td>Option of CA (option 1), PCA (2), or iterative PCA (3)
            <br />One method or another sometimes hangs due to "numerical inaccuracy."
            I subjectively try CA first, then iterative PCA, then PCA last.</td></tr>
        <tr valign="top"><td>&curren;                                      </td>
        <td><b><i>[first-view]</I></b>:                                    </td>
        <td>If this procedure file hangs or crashes after <i>N</i> reference-views, 
            <br />change the <b><i>[ca-pca]</I></b> option and 
            re-start the procedure file at <i>N+1</i>.                  </td></tr>
        <tr valign="top"><td>&curren;                                   </td>
        <td><b><i>[num-factors]</I></b>:                                </td>
        <td>Number of factors to calculate in MSA and use in classification. 
            <br />The classification does not appear to depend greatly on this value, 
            but you can check the eigenimages to see if variation is not well-represented.</td></tr>
    </table>
    

<li><p><b>Check the classification results using <font class="guitool">verifybyview</font></b></p></li>

    <p> We will check the results of the classification using 
        <font class="guitool">verifybyview.py</font>, 
        which requires that you have <a href="../../../spire/install.html"> 
        installed SPIDER's python tools</a>.
      </p>
    
    <p>Specifically, what we are looking for is that there is no subpopulation within 
       each reference-view that warrants its own class. 
       So for each reference-view, we will select the distinct class averages, and 
       use those for the next round of multi-reference alignment and classification. 
       If there is only one distinct class for each reference-view, you are finished with this step and 
       can proceed to the step of <a href="#verify">verifying individual particle images</a>.
    </p>

    <p> Open <font class="guitool">verifybyview.py</font> by typing: <br />
        <code class="snippet">verifybyview Multiref1</code>
    </p>

    <p> Included in the tarball is a starting <a href=".verifybyview">.verifybyview</a> file 
    with mostly reasonable settings. 
    However, if you viewed the outputs of <a href="binarytree.msa">binarytree.msa</a>, 
    <a href="classavg.msa">classavg.msa</a>, or <a href="kmeans.msa">kmeans.msa</a>, 
    this file will have been modified.  
<!--        Either edit the settings, or copy the archived version in the tarball, 
    called <a href=".verifybyview.multiref">.verifybyview.multiref</a>, 
    to <b>.verifybyview</b>-->
    </p>
    
    If the file <a href=".verifybyview">.verifybyview</a> is not present in your working
    directory you will need to adjust the filenames. 
    In the initial popup window, the values should be:

    <table class="outs">
        <tr valign="top"><td>&curren;</td>
            <td><font class="input">.spi</font>:</td>
            <td>Data extension.</td></tr>
        <tr valign="top"><td>&curren;</td>
            <td><font class="input">Multiref1</font>:</td>
            <td>Initial directory.</td></tr>
        <tr valign="top"><td>&curren;</td>
            <td><font class="input">listclasses</font>:</td>
            <td>List of classes.</td></tr>
        <tr valign="top"><td>&curren;</td>
            <td><font class="input">classavg***</font>:</td>
            <td>Class-average template.</td></tr>
    </table>

    <br />The first time you click on a class average, 
    a popup menu will appear to confirm filenames, 
    some of which are repeated from the initial popup menu. 
    (For this popup window, the file extensions are required, 
    but will be omitted from the list below.)
    The key files are:

    <table class="outs">
            <tr valign="top"><td>&curren;</td>
                <td><font class="input">docclass***</font>:</td>
                <td>List of particles belonging to each class.</td></tr>
            <tr valign="top"><td>&curren;</td>
                <td><font class="input">stkfiltered</font>:</td>
                <td>Filtered-image stack.</td></tr>
            <tr valign="top"><td>&curren;</td>
                <td><font class="output">goodclasses</font>:</td>
                <td>Output list of selected classes.</td></tr>
    </table>

    <p> When you click on a class average 
    (except for the first instance, when a popup menu appears), 
    a montage of individual particles will appear. 
    </p>

    <p> It is unnecessary to verify individual particles at this time. 
    Instead, simply select the class averages that you wish to use for 
    the next round of multireference alignment. 
    Select the unique classes averages by right-clicking on them. 
    Conversely, do not select similar-looking class averages, 
    which would then be used as separate references in the next round.
    When you are finished with a reference-view, save the selection by going to the 
    <i>File</I> menu and selecting <i>Save selection</I>.
    If there is only one distinct class for each reference-view, 
    proceed to the step of <a href="#verify">verifying individual particle images</a>.
    </p>

    <p> If there is an unwieldy number of classes on the main window of 
    <font class="guitool">verifybyview.py</font>, we can break the into subsets.
    Instead of the command above, type: </p>
                <code class="snippet">verifybyview Multiref1/View001</code>
                
    <p>...which will show only the classes corresponding to reference 1.  
    You can proceed to the next reference by changing the directory number (e.g., <b>View002</B>)
    and pressing <i>Update</I> (after saving the class selection, if necessary).
    </p>

<li><p><b>Iterate multireference alignment and classification</b></p></li>

    <p> We will iterate multireference alignment, filtration, and classification until 
    we obtain stable classes, <i>i.e.,</I> no new classes appear. 
    For a sample that doesn't have many orientations/conformations, 
    only 2 or 3 iterations should be needed.
    </p>
    
    The specifics in the following example are for the second round of multireference alignment. 
    The 3 steps to be iterated are:
    
      <ol>
    <li><b>Run multireference alignment</b> </li>
    
    <p> The classes selected in the previous round of multireference alignment 
        will be used as references for the next round. 
        For example, if you selected a 2 class averages from reference-view #1, 
        and one class average from reference-view #2, then 
        this next iteration will have 3 references.
    </p>
    
        <p> The input file <font class="input">docmultiref</font> 
            keeps track of successfully executed iterations of multireference alignment, and is updated. 
            The default parameter <b><i>[multirefIterNum]</I></b> of <i>-1</I> will use 
            the most recent multireference alignment as input.  
            If you do not wish to use the last multireference alignment, 
            override this parameter with the iteration number that you wish to use.
            </p>

    <li><b>Filter the aligned particles</b> </li>
    
        <p>Run <a href="filtershrink.msa">filtershrink.msa</a> as previously.</p>
    
    <li><b>Classify the particles assigned to each reference</b> </li>
    
        <p>Run <a href="classifybyview.msa">classifybyview.msa</a>, 
                set the parameter <b><i>[multirefIterNum]</I></b> if 
                you do not wish to operate on the last iteration of multireference alignment. 
                This parameter will be present in many steps from this point.</p>
                
        <li><p><b>Check the classification results using 
               <font class="guitool">verifybyview</font></b></p></li>

            <p> Select unique classes using <font class="guitool">verifybyview.py</font> 
                To open it, type: <br />
                        <code class="snippet">verifybyview Multiref2/View001</code><br />
            </p>

            Remember to update number of the directory, i.e., 
            which iteration of multireference alignment you ran.

      </ol>
      
<!--        For a possible third iteration, the following changes would be needed:
      <ol>
    <li>In <a href="multirefalign-descend.msa">multirefalign-descend.msa</a>, 
        change input file label <b><i>[old_class_dir]</I></b> from 
        <font class="input">Multiref</font> to <font class="input">Multiref2</font>
        <br />An example, <a href="multirefalign3plus.msa">multirefalign3plus.msa</a>, 
        is present in the tarball.
    </li>
    <li>In <a href="classifybyview.msa">classifybyview.msa</a>, 
        change the label <b><i>[class_dir]</I></b> from 
        <b>Multiref2</b> to <b>Multiref3</b>
    </li>
      </ol>-->
    
    <p> Iterate these 4 steps -- 
        <a href="multirefalign.msa">multirefalign.msa</a>, 
        <a href="filtershrink.msa">filtershrink.msa</a>,  
        <a href="classifybyview.msa">classifybyview.msa</a>, and
        <font class="guitool">verifybyview</font> --
        until no new classes appear. </p>
    
<!--        <p> <u>NOTE</u>: In several steps below, I will assume that you run 2 iterations of multireference alignment. 
    If you ran a different number of iterations, pay attention to when this value is required. 
    There should be reminders in the steps below when to change filenames.
    </p>-->


<li><p><a name="verify"><b>Verify individual particles from last iteration of multireference alignment</a></b></p></li>

    <p> When we compute the 3D reconstruction, we will use only the highest-quality particles. 
    To select these particles, we will use <font class="guitool">verifybyview.py</font>. 
    To start it, assuming you performed 2 rounds of multireference alignment, type: <br />
    <code class="snippet">verifybyview Multiref2/View001</code> </p>
    
    <p> You should have screened the class averages in the previous step, so 
    the filenames should be appropriate.  
    In fact, if the last directory where you viewed class averages was one of the 
    <font class="input">Multiref2/View###</font> directories, 
    you do not need to include the directory name on the command line either.
    </p>
    
    <p> Now, when you open a montage of individual particle images, 
    select the images that you want to keep using the left mouse button. 
    Conversely, exclude the particles that don't resemble the reference image. 
    To crudely assist selection of the images, the particles are 
    sorted from highest CCC to worst.  Refer to the  
    <a href="../../../spire/tools-docs/verifybyview.html">instructions for verifybyview.</a>. 
    </p>
    
    <b>Outputs:</b><br />
    
    <table class="outs">
    
    <tr valign="top"><td>&curren;</td>
        <td><font class="output">Multiref2/View###/goodclasses</font>:      </td>
        <td>List of classes with selected particles for each reference-view.</td></tr>
    <tr valign="top"><td>&curren;</td>
        <td><font class="output">Multiref2/View###/byhand***</font>:        </td>
        <td>List of selected particles for each class.                      </td></tr>
    </table>
    
    <p><u>NOTE:</u> When you finish going through the class averages for a particular reference-view, 
    be sure to save the list of good <u>classes</u> in addition to the list of good particles. 
    The good-class list can be saved by going to the <i>File</I> menu and selecting <i>Save selection</I>, 
    or by clicking on the <i>Save+Update</I> button, or with the shortcut <i>Alt-S</I>
    </p>


<li><p><b>Combine selected particles from selected classes</b></p></li>

    <p> Following the previous step, the selected particles are distributed over 
    multiple docfiles <font class="input">byhand***</font>, 
    listed in <font class="input">goodclasses</font>
    from multiple reference-views <font class="input">Multiref2/View###</font>
    (once again assuming two iterations of multireference alignment + classification). 
    In this procedure file, <a href="combinegoodclasses.msa">combinegoodclasses.msa</a>, 
    we will combine the <font class="input">byhand***</font> files into 
    one doc file for each reference-view.
    </p>
    
    <b>Output:</b><br />
    
    <table class="outs">
    
    <tr valign="top"><td>&curren;</td>
        <td><font class="output">Multiref2/goodsel***</font>:</td>
        <td>List of selected particles in each reference-view.</td></tr>
    </table>
    
<!--    <p> <i>HINT</i>: It may be useful to re-classify only the good particles, 
    if, for example, there's an asymmetry present that is washed out by the bad particles.  
    In that case, re-run classifyview, setting <b><i>[verified-yn]</I></B> to 1, 
    which will read <b>goodsel***</b> instead of <b>selview***</b>.
    You will overwrite the previous classes (but not verified particles), 
    so you may want to change the output filenames, 
    <i>e.g.</I>, add <b>good-</b> to the beginning of the filenames. 
    If you find a new class, using the names of the new doc files, re-run
    <a href="multirefalign.msa">multirefalign.msa</a>. 
    It might be useful to classify the bad particles too.
    </p>-->
    

<li><p><b>Average selected images</b></p></li>

    <p> In this step, we will generate averages of the good particle images. 
    These new averages should look cleaner than those from the multireference alignment, 
    which also included bad particles. 
    (Because of model bias, however, the bad particles may not have degraded the old average too much.)
    </p>
    
    <p> The procedure file <a href="viewaverages.msa">viewaverages.msa</a> 
    and its subroutine <a href="viewaverage1.msa">viewaverage1.msa</a> 
    read the aligned-image stack <font class="input">stk3-aligned</font> 
    and averages the images according to the good-particle selection files
    <font class="input">Multiref2/goodsel***</font> 
    (assuming again that you performed 2 rounds of multi-reference alignment).
    </p>
    
    <b>Outputs:</b><br />
    
    <table class="outs">
    
    <tr valign="top"><td>&curren;</td>
        <td><font class="output">Multiref2/stkgoodavg</font>:</td>
        <td>Stack of average images for each view.</td></tr>
    <tr valign="top"><td>&curren;</td>
        <td><font class="output">Multiref2/stkgoodvar</font>:</td>
        <td>Stack of variance images for each view.</td></tr>
    </table>


</ul>
<br />

<hr />

<p>
<b><FONT SIZE="4"><a name="mods">Recent modifications:</a></FONT></B></p>
<ul>
<!--    <li>2015-date -- </li> -->
   <li>2018-11-15 -- corrected many links, lowercase html markup al</li> 
   <li>2015-08-21 -- added "full," iterative alignment</li> 
   <li>2014-02-04 -- changed project extention to '.msa' </li>
   <li>2014-02-04 -- added SPIRE configuration file (adapted from Munich Workshop) </li>
   <li>2014-01-20 -- added more illustrations for phantom worm hemoglobin data </li>
   <li>2014-01-20 -- ca-pca.msa -- made the number of standard deviations in the factor maps user defined
      <br />&nbsp; SPIDER sometimes crashes when too small a value is used 
      (where too many points pile up on the edge of the plot)</li>
   <li>2009-06-12 -- added illustration for custom mask</li>
</ul>

<hr />

<b><p><a name="refs">References</a></p></b>
<ol>
    <a name="jf2006">
    <li>Frank J. (2006) 
        <i>Three-Dimensional Electron Microscopy of Macromolecular Assemblies</I>.
        (Oxford University Press, New York, NY).</li>
    </a>
    <li>Marco S, Chagoyen M, de la Fraga LG, Carazo JM, Carrascosa JL. (1996) 
        A variant to the random approximation of the reference-free algorithm.
        <i>Ultramicroscopy</I> <b>66</b>: 5-10.</li>
<!--    <li>AUTHOR 1, AUTHOR 2. (YEAR) 
        TITLE.
        <i>JOURNAL</I> <b>VOLUME</b>: PG0-PG.</li>-->
    <li>Shaikh TR, LeBarron JS, Trujillo R, Baxter WT, and Frank J. (2008a) 
        Particle-verification for single-particle reconstruction using multivariate data analysis and classification.
        <i>J Struct Biol </I> <b>161</b>: 41-48.</li>
    <li>Shaikh TR, Gao H, Baxter WT, Asturias F, Boisset N, Leith A, and Frank J. (2008b) 
        SPIDER image-processing for single-particle reconstruction of biological macromolecules 
        from electron micrographs.
        <i>Nature Protocols</I> <b>3</b>: 1941-74.</li>
    <li>Huang T, Shaikh TR, Gupta K, Contreras-Martin LM, Grassucci RA, Van Duyne GD, Frank J, Belfort M .
        (2011) The group II intron ribonucleoprotein precursor is a large, loosely packed structure. 
        <i>Nucleic Acids Res.</I> <b>39</b>: 2845-54.</li>
</ol>

<hr />

<p>
<small>
  Source:       spider/docs/techs/MSA/index.html &nbsp;&nbsp;&nbsp; 
  Page updated: 2018/11/15           &nbsp;&nbsp;&nbsp; 
  Tanvir Shaikh
</small>
</p>

</body>
</html>
